Create testing for cluster-related functions and add Fluidigm data

catterbu opened the pull request:
The validation functions, the `multi_clust` function, and all cluster plotting functions have tests to ensure that they are not broken in future updates.

catterbu committed:
Fix data.frame not being used in arg validation test
  - Fixes #21

catterbu committed:
Attempt to add hashes
  - Add set.seed to test multi_clustering and cluster_plotting
  - Fix hashes running all code in test_cluster_plotting
  - More hash fixing
  - Add iris dataset for testing experimentation
  - Change hashes for iris use
  - Remove library datasets
  - Change wss_plot test to use file size for hash
  - Change clusGap_plot test to use file size for hash
  - Change pca_plot test to use file size for hash
  - Change sil_plot test to use file size for hash
  - Change avg_sil_plot test to use file size for hash
  - Change clust_boxplot test to use file size for hash
  - Change hashes for new set.seed
  - Change to using png with height/width specs

catterbu committed:
Use transform in clust_boxplot

catterbu committed:
Add multi_clust testing

nsh87 commented:
This works:

``` R
context("Unit test cluster plotting")

tryCatch({
    set.seed(1)
    data(fluidigm)
    fluidigm <- fluidigm[1:25,]
    f_clust <- multi_clust(fluidigm, krange = 2:4)
  },
  finally = {
    set.seed(NULL)  # Turn off seed
  }
)

test_that("wss plots properly", {
  expected_hash <- "47d160c26197e5ef76cd8d0416f56774"
  file_path <- tempfile(pattern="wss_plot", tmpdir=tempdir(), fileext=".png")
  png(file_path, width=480, height=480, antialias="none")
  wss_plot(f_clust, optimal=FALSE)
  dev.off()
  test_hash <- digest::digest(file.info(file_path)[["size"]], algo="md5")
  expect_that(expected_hash, equals(test_hash))
})
```

catterbu committed:
Remove image testing in cluster_plotting

catterbu commented:
@nsh87 Assuming nothing goes wrong with travis, this should be ready to merge now.

catterbu committed:
Add fluidigm data set and documentation

catterbu committed:
Fix travis errors
  - Remove trailing white space in fluidigm docs
  - Fix missing quote character in multi_clust validation
  - Fix clusGap refs to gap

catterbu committed:
Fix travis error with classes of vectors

catterbu commented:
@nsh87 I am hesistant to write anything after it got rejected before for something that I thought I had already fixed, but this passed the tests on my local. If travis is kind, this should be ready.

catterbu committed:
Add fluidigm Rd file
